pkg/controller/storagecluster/storagecluster_controller_test.go

```diff
@@ -759,3 +759,149 @@ func TestNodeTopologyMapLabelSelector(t *testing.T) {
+	  - "namespace"
+	  "group_interval": "5m"
+	  "group_wait": "30s"
+	  "receiver": "Default"
+	  "repeat_interval": "12h"
+	  "routes":
+	  - "match":
+	      "alertname": "Watchdog"
+	    "receiver": "Watchdog"
+	  - "match":
+	      "severity": "critical"
+	    "receiver": "Critical"
+		`
+	configWithInhibition := `
+	"global":
+	  "resolve_timeout": "5m"
+	"inhibit_rules":
+	- "equal":
+	  - "namespace"
+	  - "alertname"
+	  "source_match_re":
+	   "poddisruptionbudget" : "^rook-ceph-osd-[0-9]+$"
+	  "target_match_re":
+	    "poddisruptionbudget" : "^rook-ceph-osd-[0-9]+$"
+	- "equal":
+	  - "namespace"
+	  - "alertname"
+	  "source_match":
+	    "severity": "critical"
+	  "target_match_re":
+	    "severity": "warning|info"
+	- "equal":
+	  - "namespace"
+	  - "alertname"
+	  "source_match":
+	    "severity": "warning"
+	  "target_match_re":
+	    "severity": "info"
+	"receivers":
+	- "name": "Default"
+	- "name": "Watchdog"
+	- "name": "Critical"
+	"route":
+	  "group_by":
+	  - "namespace"
+	  "group_interval": "5m"
+	  "group_wait": "30s"
+	  "receiver": "Default"
+	  "repeat_interval": "12h"
+	  "routes":
+	  - "match":
+	      "alertname": "Watchdog"
+	    "receiver": "Watchdog"
+	  - "match":
+	      "severity": "critical"
+	    "receiver": "Critical"
+		`
+	alertmanagerConfigSecretName := "alertmanager-main"
+	alertmanagerConfigSecretNamespace := "openshift-monitoring"
+	secretWithoutInhibition := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alertmanagerConfigSecretName,
+			Namespace: alertmanagerConfigSecretNamespace,
+		},
+		Data: make(map[string][]byte, 0),
+	}
+	secretWithInhibition := secretWithoutInhibition.DeepCopy()
+	secretWithoutInhibition.Data["alertmanager.yaml"] = []byte(configWithoutInhibition)
+	secretWithInhibition.Data["alertmanager.yaml"] = []byte(configWithInhibition)
+	type args struct {
+		actualSecret         *corev1.Secret
+		expectedSecret       *corev1.Secret
+		storageClusterLength int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "No inhibition rule, no Storagecluster",
+			args: args{
+				actualSecret:         secretWithoutInhibition,
+				expectedSecret:       secretWithoutInhibition,
+				storageClusterLength: 0,
+			},
+		},
+		{
+			name: "No inhibition rule, 1 Storagecluster",
+			args: args{
+				actualSecret:         secretWithoutInhibition,
+				expectedSecret:       secretWithInhibition,
+				storageClusterLength: 1,
+			},
+		},
+		{
+			name: "Inhibition rule, no Storagecluster",
+			args: args{
+				actualSecret:         secretWithInhibition,
+				expectedSecret:       secretWithoutInhibition,
+				storageClusterLength: 0,
+			},
+		},
+		{
+			name: "Inhibition rule, 1 Storagecluster",
+			args: args{
+				actualSecret:         secretWithInhibition,
+				expectedSecret:       secretWithInhibition,
+				storageClusterLength: 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Why are you doing a `t.Run()` here? I don't see that we do it for any of the other tests. I might just be ignorant, though.

Reply:
This is auto-generated. Can be used to run tests in parallel.
Since, we're not using it anywhere else, dropping it for now.